packaging/linux/postinstall

```diff
@@ -1,15 +1,21 @@
 #!/bin/bash
 
+set -o errexit
+
 user=$(loginctl list-users --output json | jq -r ".[0].user")
 home=$(getent passwd "${user}" | cut -d: -f 6)
 
 config_dir="${home}/.config/naisdevice/"
-unit_file=/lib/systemd/system/device-agent-helper.service
+unit_file=/usr/lib/systemd/system/device-agent-helper.service
```

Comment:
Ah.. interesting. Was this because it landed there on your machine? The location is decided by dpkg as far as I can tell (or by fpm when you inform it about a systemd-file to add).
The location might be system dependent, the original path is where it landed on my machine. In that case, we need to query the system to find the file.

Reply:
Both exist on my system - so I think maybe we should follow the (presumably sane) defaults of FPM or double-check before insisting on own changes? But then again, maybe @Sechmann knows much more than me wrt. systemd FHS layout.

Reply:
On ubuntu-20.10 `/lib` is just a symlink to `/usr/lib`, so i just thought it would make more sense to reference the real path instead of the symlink

Reply:
As for the daemon-reload, will fpm reload before or after postinstall?

The reason i put this here was because i noticed the loaded system file contained the @@CONFIG_DIR@@ which should have been replaced, however i see now that the actual file on disk also contains this.

```
$ cat /usr/lib/systemd/system/device-agent-helper.service
[Unit]
Description=Naisdevice agent helper
Documentation=https://doc.nais.io/device

[Service]
ExecStart=/usr/sbin/device-agent-helper --config-dir @@CONFIG_DIR@@
Restart=always

[Install]
WantedBy=multi-user.target
```

So I'm guessing the problem is with the sed command then

Reply:
I found this: (https://wiki.debian.org/systemd#:~:text=Unit%20files%20provided%20by%20Debian,lib%2Fsystemd%2Fsystem%20directory.)

> Unit files provided by Debian are located in the /lib/systemd/system directory.

So the "canonical" path is under `/lib/`, which happens to be a symlink on Ubuntu. I think it's better to refer to the "canonical" path.